### PR TITLE
Updated the README

### PR DESCRIPTION
**Pull request by nathannaveen**
* Included what a roaring bitmap is.
* Included a highlevel overview diagram from the users perspective

## Summary by CodeRabbit

- **New Features**
	- Enhanced the README.md to provide a detailed description of the BitBom Minefield tool, highlighting its use of roaring bitmaps for graphing Software Bill of Materials (SBOMs).
	- Added a section explaining what a roaring bitmap is, improving educational value with external references.
	- Introduced an "Overview of BitBom" section with a sequence diagram to illustrate user interactions and processes related to the Minefield tool.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,36 @@
 # Minefield
 
-Minefield is a tool for graphing Software Bill of Materials (SBOMs) and other types of metadata in a highly extensible and efficient graph.
+BitBom Minefield is a tool that uses roaring-**Bit**maps to graph S**BOM**s.
```
[the remark this answers is not in the document]
Comment (nathannaveen) [reply README.md:3]: This is intentional

[the remark this answers is not in the document]
Comment (nathannaveen) [reply README.md:3]: "roaring-**Bit**maps" shouldn't be used everywhere, just in this instance.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,36 @@
...
+
+## What is a Roaring Bitmap
+
+[Roaring bitmaps](https://github.com/RoaringBitmap/roaring) is a data structure that creates ginormous bitmaps.
```
Comment (neilnaveen) [inline README.md:7]: could the is be changed to an are

**Review by neilnaveen — COMMENTED** [review]
Other tha this everything looks good

**Review by naveensrinivasan — APPROVED** [review]

**Pull request merged**